enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1934,9 +1943,14 @@ func (p *PebbleCache) newCDCCommitedWriteCloser(ctx context.Context, fileRecord
 }
 
 func (cdcw *cdcWriter) writeChunk(chunkData []byte) error {
+	cdcw.mu.Lock()
```

Comment:
IIUC, there's no real concurrency to worry about since writeChunk is only called by 1 goroutine and Commit will only need to access the mutex protected data when writeChunk is guaranteed to not be called. I think this means you could get away with locking the mutex for the entire duration of writeChunk, i.e.

```
cdcw.my.Lock()
defer cdcw.mu.Unlock()
``` 
at the top of the function

I think you can do something similiar in Commit after `cdcw.eg.Wait()`